src/libraries/System.Diagnostics.DiagnosticSource/src/System.Diagnostics.DiagnosticSource.csproj

```diff
@@ -19,15 +20,15 @@
     <DefineConstants Condition="'$(TargetFramework)' != 'netstandard1.1' and '$(TargetFramework)' != 'netstandard1.3'">$(DefineConstants);EVENTSOURCE_ENUMERATE_SUPPORT</DefineConstants>
     <DefineConstants Condition="$(TargetFramework.StartsWith('net4'))">$(DefineConstants);ALLOW_PARTIALLY_TRUSTED_CALLERS;ENABLE_HTTP_HANDLER</DefineConstants>
     <ExcludeFromPackage Condition="'$(TargetFramework)' == 'netstandard2.0'">true</ExcludeFromPackage>
-    <DefineConstants Condition="'$(TargetFramework)' == '$(NetCoreAppCurrent)'">$(DefineConstants);W3C_DEFAULT_ID_FORMAT</DefineConstants>
+    <DefineConstants Condition="$([MSBuild]::GetTargetFrameworkIdentifier('$(TargetFramework)')) == '.NETCoreApp'">$(DefineConstants);W3C_DEFAULT_ID_FORMAT</DefineConstants>
```

Comment:
>GetTargetFrameworkIdentifier [](start = 45, length = 28)

what happen when .NET 5.0 is the target framework? this condition would be false here?

Reply:
The TargetFrameworkIdentifier is `.NETCoreApp` for `.NET 5.0+`. It made sense to leave it that way I guess because using `.NET Framework` would be true for full framework.

Reply:
thanks!

In .NET 7, would we add .net6.0 to the targets framework? I am just trying to understand the how stuff work in general in the future.  
Also, what happen if we didn't specify .net5.0 in the target frameworks? would ns2.0 would be picked for .net5.0?

---
In reply to: [577102816](https://github.com/dotnet/runtime/pull/48318#discussion_r577102816) [](ancestors = 577102816)

Reply:
> Also, what happen if we didn't specify .net5.0 in the target frameworks? would ns2.0 would be picked for .net5.0?

The nearest compatible tfm would be picked up which in this case would be netstandard2.0 if we wouldn't have `$(NetCoreAppCurrent)` or `net5.0` configurations, right.

> In .NET 7, would we add .net6.0 to the targets framework? I am just trying to understand the how stuff work in general in the future.

You would only add net6.0 as a configuration if you want to use APIs that aren't available in net5.0.

Reply:
one last question, does the condition Condition="$([MSBuild]::GetTargetFrameworkIdentifier('$(TargetFramework)')) == '.NETCoreApp'" will be true for .NET Core 3.x and 2.x?

---
In reply to: [577121637](https://github.com/dotnet/runtime/pull/48318#discussion_r577121637) [](ancestors = 577121637)

Reply:
Tarek you can ask as many questions as you want :)

Yes, the TargetFrameworkIdentifier is ".NETCoreApp" and applies to netcoreapp1.x, netcoreapp2.x, netcoreapp3.x, net5.0, net6.0, > net6.0. The TargetFrameworkIdentifier for .NET Framework is ".NETFramework" and applies to net1.x, net2.x, net3.x, net4.x. The TargetFrameworkIdentifier for .NET Standard is ".NETStandard" and applies to netstandard1.x and netstandard2.x.

Hope that makes it clearer :)

Reply:
Thanks Viktor! 

I would suggest you create a property for the condition $([MSBuild]::GetTargetFrameworkIdentifier('$(TargetFramework)')) != '.NETCoreApp'" instead of repeating the condition many times in the project. 

---
In reply to: [577155551](https://github.com/dotnet/runtime/pull/48318#discussion_r577155551) [](ancestors = 577155551)

Reply:
I could only define such a property in the project directly as the `$(TargetFramework)` property doesn't exist before the project is evaluated (i.e. in a props file). Doing that would be repetitive as I would need to define it in every project where I want to check on the TargetFrameworkIdentifier. The recommendation is to use such properties instead where they need to be conditioned so I think we just need to live with that long syntax...